monkey/tests/unit_tests/agent_plugins/exploiters/snmp/test_snmp_plugin.py

```diff
@@ -0,0 +1,190 @@
+
+
+@pytest.fixture
+def open_snmp_ports():
+    return TargetHostPorts(
+        udp_ports=PortScanDataDict(
+            {p: PortScanData(port=p, status=PortStatus.OPEN) for p in SNMP_PORTS}
+        )
+    )
+
+
+@pytest.fixture
+def target_host(open_snmp_ports) -> TargetHost:
+    return TargetHost(
+        ip=TARGET_IP,
+        operating_system=OperatingSystem.LINUX,
+        ports_status=open_snmp_ports,
+    )
+
+
+class ErrorRaisingMockSNMPExploiter(SNMPExploiter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def exploit_host(self, *args, **kwargs) -> ExploiterResult:
+        raise Exception("Test error")
+
+
+@pytest.fixture
+def mock_snmp_exploiter():
+    exploiter = MagicMock(spec=SNMPExploiter)
+    exploiter.exploit_host.return_value = EXPLOITER_RESULT
+    return exploiter
+
+
+@pytest.fixture
+def plugin(monkeypatch, mock_snmp_exploiter) -> Plugin:
+    monkeypatch.setattr(
+        "agent_plugins.exploiters.snmp.src.plugin.SNMPExploiter",
+        lambda *args, **kwargs: mock_snmp_exploiter,
+    )
+    monkeypatch.setattr(
+        "agent_plugins.exploiters.snmp.src.plugin.SNMPClient",
+        lambda *args, **kwargs: MagicMock(spec=SNMPClient),
+    )
+
+    return Plugin(
+        plugin_name="SNMP",
+        agent_id=AGENT_ID,
+        agent_event_publisher=MagicMock(),
+        agent_binary_repository=MagicMock(),
+        http_agent_binary_server_registrar=MagicMock(),
+        propagation_credentials_repository=MagicMock(spec=IPropagationCredentialsRepository),
+        local_machine_info=MagicMock(),
+        agent_command_builder_factory=MagicMock(),
+    )
+
+
+def test_run__fails_on_bad_options(plugin: Plugin, target_host: TargetHost):
+    result = plugin.run(
+        host=target_host,
+        options=BAD_SNMP_OPTIONS_DICT,
+        interrupt=Event(),
+    )
+
+    assert not result.exploitation_success
+    assert not result.propagation_success
+
+
+def test_run__attempts_exploit_if_port_status_unknown(
+    plugin: Plugin,
+    mock_snmp_exploiter,
+    target_host: TargetHost,
+):
+    host = target_host
+    host.ports_status.udp_ports = PortScanDataDict({})
+    result = plugin.run(
+        host=host,
+        options={},
+        interrupt=Event(),
+    )
+
+    mock_snmp_exploiter.exploit_host.assert_called_once()  # type: ignore [attr-defined]
+    assert result == EXPLOITER_RESULT
+
+
+def test_run__attempts_exploit_if_port_status_open(
+    plugin: Plugin,
+    mock_snmp_exploiter,
+    target_host: TargetHost,
+):
+    host = target_host
+    host.ports_status.udp_ports = PortScanDataDict(
+        {SNMP_PORTS[0]: PortScanData(port=SNMP_PORTS[0], status=PortStatus.OPEN)}
+    )
+    result = plugin.run(
+        host=host,
+        options={},
+        interrupt=Event(),
+    )
+
+    mock_snmp_exploiter.exploit_host.assert_called_once()  # type: ignore [attr-defined]
+    assert result == EXPLOITER_RESULT
```

Comment:
NIT: You could parametrize these. I'm fine with them being separate, though. Either way.